packages/common/src/signHelpers.ts

```diff
@@ -3,40 +3,175 @@ import {
+ * A structure to use for creating signatures on a given digest.
+ */
 export interface AccountSigner {
+    /**
+     * Creates a signature of the provided digest
+     *
+     * @param {Buffer} digest - The digest to create signatures on.
+     *
+     * @returns {Promise<AccountTransactionSignature>} A promise resolving with a set of signatures for a set of credentials corresponding to some account
+     */
     sign(digest: Buffer): Promise<AccountTransactionSignature>;
+    /**
+     * Amount of signatures created
+     */
     getSignatureCount(): bigint;
 }
 
+const getSignature = async (
+    digest: Buffer,
+    privateKey: string
+): Promise<HexString> =>
+    Buffer.from(await ed.sign(digest, privateKey)).toString('hex');
+
 /**
- * Creates a signer for an account which uses the first credential's first keypair.
+ * Creates an `AccountSigner` for an account which uses the first credential's first keypair.
  * Note that if the account has a threshold > 1 or the first credentials has a threshold > 1, the transaction signed using this will fail.
- * @param privateKey the ed25519 private key in HEX format. (First credential's first keypair's private key)
+ *
+ * @param {HexString} privateKey - the ed25519 private key in HEX format. (First credential's first keypair's private key)
+ *
+ * @returns {AccountSigner} an `AccountSigner` which creates a signature using the first credentials first keypair
  */
 export function buildBasicAccountSigner(privateKey: string): AccountSigner {
     return {
         getSignatureCount() {
             return 1n;
         },
         async sign(digest: Buffer) {
-            const signature = Buffer.from(
-                await ed.sign(digest, privateKey)
-            ).toString('hex');
             return {
                 0: {
-                    0: signature,
+                    0: await getSignature(digest, privateKey),
                 },
             };
         },
     };
 }
 
+const isWalletExport = <T extends WithAccountKeys>(
+    value: T | WalletExportFormat
+): value is WalletExportFormat =>
+    (value as WalletExportFormat).value?.accountKeys !== undefined;
+
+const isSimpleAccountKeys = <T extends WithAccountKeys>(
+    value: T | WalletExportFormat | SimpleAccountKeys
+): value is SimpleAccountKeys =>
+    (value as WalletExportFormat).value?.accountKeys === undefined &&
+    (value as T).accountKeys === undefined;
+
+const getKeys = <T extends WithAccountKeys>(
+    value: T | WalletExportFormat | SimpleAccountKeys
+): SimpleAccountKeys => {
+    if (isSimpleAccountKeys(value)) {
+        return value;
+    }
+    const { keys } = isWalletExport(value)
+        ? value.value.accountKeys
+        : value.accountKeys;
+
+    return Object.entries(keys).reduce(
+        (acc, [k, v]) => ({
+            ...acc,
+            [k]: Object.entries(v.keys).reduce(
+                (a, e) => ({ ...a, [e[0]]: e[1].signKey }),
+                {}
+            ),
+        }),
+        {}
+    );
```

Comment:
```suggestion
    return mapRecord(keys, (credKeys) => mapRecord(credKeys.keys, (keyPair) => keyPair.signKey));
```
I think. If you are okay with using existing helper functions :thinking: